Payjoin-spec.md

```diff
@@ -0,0 +1,61 @@
+# Payjoin in BTCPay Server
+
+## Introduction
+
+Payjoin is a generic term for any privacy mechanism where a merchant can protect his own privacy and the privacy of his customers by mixing his own UTXO to the payment of the customer.
+
+Bustapay is an specification of such mechanism. We identified some issues with the specification so we needed to deviate from it.
+
+## Difference with Bustapay
+
+The [bustapay (BIP79)](https://github.com/bitcoin/bips/blob/master/bip-0079.mediawiki) proposal has been written for payjoin.
+
+We initially implemented it, but found out that the specification was missing certain important aspect.
+Also, because this is a new proposal, we expect that even our own proposal will experience breaking changes in the following week and months.
+As such, we preferred writting our own specification for the time being.
+Once the specification stabilize and have gone through review and testing by the community, we will decide whether we will submit to the BIP process, or if we can just modify bustapay BIP79.
+
+Bustapay is currently not widely deployed, so we are not worried about breaking backward compatibility.
+
+Our implementation is identical to bustapay with the following differences.
+```diff
+- Step 1. Sender creates a bitcoin transaction paying the receiver
++ Step 1. Sender creates a PSBT paying the receiver
```

Comment:
I'm still not completely sold on completely dropping support for accepting raw transaction format. We only gain fee rate checking on transaction submission to the endpoint for much less integration potential.

Reply:
All languages have libraries with PSBT support.

Reply:
Yeah but  lot of wallets barely managed to tape together segwit support, much less psbt support.

Reply:
those wallet won't touch payjoin :p